crypto/fipsmodule/ec/ec_key.c

```diff
@@ -381,6 +382,26 @@ int EC_KEY_check_fips(const EC_KEY *key) {
     return 0;
   }
 
+  // Check that the coordinates are within the range [0,p-1], when the (raw)
+  // point is affine; i.e. Z=1.
+  // This is the case when validating a received public key.
+  // Note: The check for x and y being negative seems superfluous since
+  // ec_felem_to_bignum() calls BN_bin2bn() which sets the `neg` flag to 0.
+  if(ec_felem_equal(key->pub_key->group, &key->pub_key->group->one, &key->pub_key->raw.Z)) {
+    x = BN_new();
```

Comment:
I don't see where you clean up the heap so you will be leaking memory with each check you execute.  You will need to free with the return 0; and also the case where the test passes.